internal/runners/initialize/init.go

```diff
@@ -103,11 +106,55 @@ func prepare(params *RunParams) error {
 
 // Run kicks-off the runner.
 func (r *Initialize) Run(params *RunParams) error {
-	_, err := run(params, r.Outputer)
+	_, err := run(params, r.out, r.proj)
 	return err
 }
 
-func run(params *RunParams, out output.Outputer) (string, error) {
+func runHeadless(params *RunParams, out output.Outputer, proj *project.Project) (string, error) {
+	if params.Language != "" {
+		return "", locale.NewInputError("init_headless_lang_provided_err", "You should not provide a language, dude!")
+	}
+
+	headlessCommitID := proj.CommitID()
+
+	err := proj.Source().SetNamespace(params.Namespace.Owner, params.Namespace.Project)
+	if err != nil {
+		return "", errs.Wrap(err, "Could not set namespace in project file.")
+	}
+
+	fail := proj.Source().SetCommit(headlessCommitID, false)
+	if fail != nil {
+		return "", errs.Wrap(fail.ToError(), "Could not set commit id in project file.")
+	}
+
+	path := params.Path
+	if path == "" {
+		path, err = osutils.Getwd()
+		if err != nil {
+			return "", errs.Wrap(err, "Could not determine current working directory.")
+		}
+	}
+
+	out.Notice(locale.Tr(
+		"init_success",
+		params.Namespace.Project, params.Namespace.Owner, path))
+	return path, nil
+}
+
+func run(params *RunParams, out output.Outputer, proj *project.Project) (string, error) {
+	// try to initialize project at --path if provided
+	if params.Path != "" {
+		var fail *failures.Failure
+		proj, fail = project.Parse(filepath.Join(params.Path, constants.ConfigFileName))
+		if fail != nil {
+			logging.Debug("No project file parsed at path: %v", fail.ToError())
```

Comment:
You are assuming a failure means that the file doesn't exist. Please be explicit.